### PR TITLE
Bootstrap tables and buttons

### PR DESCRIPTION
**Pull request by jir-f**
Added bootstrap buttons and tables

Pull Request Description:
Added bootstrap buttons with block syntax support.
Added bootstrap tables with block syntax support.

New Pull Request Checklist:
- [X] Reviewed [contributing guidelines](CONTRIBUTING.md).
- [X] [good commit messages](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
- [] Github issue added, if it exists.
- [X] Added an entry to the [Changelog](CHANGELOG.md).
- [X] All tests passing: run `npm test`.
- [X] New component tests include [ember-a11y-testing](https://github.com/ember-a11y/ember-a11y-testing) to test accessibility.

### CODE REVIEW

**Comment by jweakley** [comment]
See https://ember-bootstrap-controls.herokuapp.com/

**Comment by flarinerin** [comment]
Screen Shot 2019-04-01 at 11 49 04 AM

changing values in component options appears to not do anything to the component displayed.

Having tabular data that forces the table to be wider than the small screen would be better to show how table responsive works.

**Comment by flarinerin** [comment]
@jweakley can you redeploy with
 "Updated how freestyle tables is displayed. Fixed checkbox bug."
fixes in it so I can verify that it's fixed?

**Comment by jweakley** [comment]
@flarinerin this PR has been redeployed

**Comment by flarinerin** [comment]
I recommend changing the table data to much longer examples so that table-responsive is needed and can display as expected. The class is being applied as expected but general users, new team members, and guests using our library will not be able to know the difference.

**Comment by jweakley** [comment]
@flarinerin more data was added to the table and deployed for you to review.

File: addon/templates/components/bootstrap/tables/-pagination-control.hbs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{{#if (or (and (eq totalPages 1) (gt pageRecordsMeta.countTotal 5)) (and (gt totalPages 1) pageRecordsMeta))}}
```
[the remark this answers is not in the document]
Comment (harokevin) [reply addon/templates/components/bootstrap/tables/-pagination-control.hbs]: by controller I just mean the js for this component. Sorry for the confusion.

**Comment by harokevin** [comment]
Updates^^

File: addon/templates/components/bootstrap/tables/-pagination-control.hbs (hunk 1 above)
[the remark this answers is not in the document]
Comment (harokevin) [reply addon/templates/components/bootstrap/tables/-pagination-control.hbs]: Done

**Comment by harokevin** [comment]
When working on moving the logic from `addon/templates/components/bootstrap/tables/-pagination-control.hbs` into the js I noticed that there is a bug with displaying the pagination on the `table-with-pagination`. It currently renders in freestyle but does not show any content. 

It seems like it does not show any content because the conditional on line 1 of `ember-bootstrap-controls/addon/templates/components/bootstrap/tables/-pagination-control.hbs` which is now `showPaginationNav` resolves to false. I it seems like it resolves to false because pageRecordsMeta seems to be undefined and is required in all cases for the `showPaginationNav` flag to become true.

I did some more looking and it seems like `pageRecordsMeta` is later set as a property on the component as a result of `calculatePaginationTask` running inside of `didReceiveAttrs` in `ember-bootstrap-controls/addon/components/bootstrap/tables/table-with-pagination.js`. There could be an asycn bug where where the 

```
  {{#if paginationEnabled}}
    {{bootstrap/tables/-pagination-control
      pageNumber=pageNumber
      pageSize=pageSize
      totalPages=totalPages
      pageRecordsMeta=pageRecordsMeta
      onPage=(action 'onPage')
    }}
  {{/if}}
```
part of  `table-with-pagination` is getting the `pageRecordsMeta` before the parent component has time to define it.

**Comment by harokevin** [comment]
I did some more digging and is also seems that -- at least in freestyle -- line 86 of `table-with-pagination` does not resolve because 1) `rowsData` is not an object, it is an array, and 2) because meta.page does not exist on rowsData. `const paginationInformation = yield get(rowsData, 'meta.page');`. Either the seed data given to the component is incorrect (I am assuming it is because we are trying to showcase pagination in freestyle ) or the component is behaving appropriately and there is something I am missing.

**Comment by jweakley** [comment]
The seed data should be an ember data collection. I think we should make it be able to handle an array as well though. The problem is that arrays would not have any of the meta paging data so pagination controls don’t really make sense for an array. We might want to have the table work with arrays and have the wrapper be able to pick apart an ember data array.

**Comment by jir-f** [comment]
I changed the rowData that is being passed to the table to be a ArrayProxy. The models have a meta object that has the pagination information. 
I also tested it with passing only an array of objects with no meta, and the table was rendered without the pagination components.

**Comment by harokevin** [comment]
If I change the `per page` dropdown to 5 should I expect the table to only show 5 rows? @jir-f 
It currently does not adjust in this way.

**Comment by Samsinite** [comment]
Please read feedback

**Comment by harokevin** [comment]
These are issues that would be nice to address in the future but are not blockers
https://github.com/wildland/ember-bootstrap-controls/issues/247
https://github.com/wildland/ember-bootstrap-controls/issues/244
https://github.com/wildland/ember-bootstrap-controls/issues/243

**Pull request merged**